packages/eslint-plugin/src/rules/no-base-to-string.ts

```diff
@@ -184,6 +187,10 @@ export default createRule<Options, MessageIds>({
         return Usefulness.Always;
       }
 
+      if (checker.isArrayType(type) || checker.isTupleType(type)) {
+        return collectArrayToStringCertainty(type);
```

Comment:
It's convenient to reuse `collectArrayToStringCertainty` but it does mean some checks (like `checker.isArrayType`, `checker. isTupleType `, but also union/intersection checks run multiple times).